finagle-http/src/main/scala/com/twitter/finagle/http/HeaderMap.scala

```diff
@@ -52,67 +54,88 @@ abstract class HeaderMap
   def += (kv: (String, Date)): HeaderMap =
     += ((kv._1, HeaderMap.format(kv._2)))
 
-  override def empty: HeaderMap = new MapHeaderMap(mutable.Map.empty)
+  override def empty: HeaderMap = MapHeaderMap()
 }
 
+private[finagle] case class HeaderValuePair(header: String, value: String) {
+  val canonicalName = HeaderValuePair.canonicalName(header)
+}
+
+private[finagle] object HeaderValuePair {
+  def canonicalName(header: String) = header.toLowerCase
```

Comment:
I was aware of that point, I saw no harm in leaving it to the default because I think as long as it is consistent through out the life cycle of the running app (or most likely within the life cycle of a single request) it *should* be fine. So that was my reasoning.

However I am no expert on this stuff, so tried to add what you suggested. It didn't work out because it expects a `Locale` not a `Charset`. I ended up using `Locale.US`. If there is a twitter convention for this please let me know.